cgp/genome.py

```diff
@@ -346,6 +349,10 @@ def _is_active_input_gene(self, gene_idx: int) -> bool:
         else:
             assert False  # should never be reached
 
+    def _select_gene_indices(self, mutation_rate, dna):
```

Comment:
could you please add annotations? and maybe rename the function to `_select_gene_indices_for_mutation`? and maybe you don't need to pass the dna, just the length of it since you never use it besides in `len(dna)`